src/main/scala/chiffre/util/ScanChainConfig.scala

```diff
@@ -48,15 +48,17 @@ class ScanChainUtils(implicit opt: Arguments) {
 
   private def bind(f: ScanField, name: String)(implicit opt: Arguments): ScanField = try {
     f match {
-        case x: ScanField if x.value.nonEmpty => throw new ScanChainException(
-          s"Tried to rebind already bound ScanField $f (name: $name)")
-        case x: Seed        => x.bind(BigInt(x.width, rand))
-        case x: Difficulty  => x.bind(opt.probability)
-        case x: Mask        => x.bind(opt.mask)
-        case x: StuckAt     => x.bind(opt.stuckAt)
-        case x: Cycle       => x.bind(opt.cycle)
-        case x: CycleInject => x.bind(opt.cycleInject)
-        case _              => throw new ScanChainException(s"Unimplemented binding for ScanField $f")
+      case x: ScanField if x.value.nonEmpty =>
+        if (opt.errorIfBound) {
+          throw new ScanChainException(s"Tried to rebind already bound ScanField $f (name: $name)") }
+        x
+      case x: Seed        => x.bind(BigInt(x.width, rand))
+      case x: Difficulty  => x.bind(opt.probability)
+      case x: Mask        => x.bind(opt.mask)
+      case x: StuckAt     => x.bind(opt.stuckAt)
+      case x: Cycle       => x.bind(opt.cycle)
+      case x: CycleInject => x.bind(opt.cycleInject)
+      case _              => throw new ScanChainException(s"Unimplemented binding for ScanField $f")
```

Comment:
Since the JSON now has those `null` fields for users to fill out, isn't this redundant?  It will need to be removed anyway by #19.

Reply:
There are two ways of binding values in a scan field provided by this PR with decreasing precedence:

  1. A user manually updates values in the JSON
  2. Any unbound value in the scan chain can get a default value provided via a command line argument

That first case will cause a bound value to never be overwritten by one of the CLI arguments (though I should simplify it to `case x: ScanField if x.isBound =>`). It would be clearer if I updated the CLI API to be `--default-mask`, etc.

For this reason, I think that this enables what you need with #19, though there currently exists no CLI API that lets you bind a specific scan field instance (only all scan field types). If the user wants this behavior, they need to go and edit the JSON.

Does that make sense to you?

Reply:
I see.  Yes, that makes sense.

Reply:
Thanks for checking my logic here! There is [one new test case](https://github.com/IBM/chiffre/pull/21/files#diff-1a5254651c1372f08ff90dab517502aaR75) that a bound field can't be overwritten by an existing CLI argument.